subsys/fs/littlefs_fs.c

```diff
@@ -680,18 +761,40 @@ static int littlefs_mount(struct fs_mount_t *mountp)
 		 "cache size incompatible with block size");
 
 	/* Set the validated/defaulted values. */
-	lcp->context = (void *)fs->area;
-	lcp->read = lfs_api_read;
-	lcp->prog = lfs_api_prog;
-	lcp->erase = lfs_api_erase;
+	if (littlefs_on_blkdev(mountp)) {
+#ifdef CONFIG_FS_LITTLEFS_BLK_DEV
+		lcp->context = (void *)mountp->storage_dev;
+		lcp->read = lfs_api_read_blk;
+		lcp->prog = lfs_api_prog_blk;
+		lcp->erase = lfs_api_erase_blk;
+
+		lcp->read_size = block_size;
+		lcp->prog_size = block_size;
+		lcp->cache_size = block_size;
+		lcp->lookahead_size = block_size * 4;
+
+		erase_buffer = fc_allocate(block_size);
```

Comment:
That was somewhat a "hack" for this RFC. The idea, which goes over my mind is to use the in-flash buffer with 0xFFs (allocated with linker script).

Reply:
Shouldn't the `erase` of a sdmmc just return 0 as the erase is done automatically before a write ?

Reply:
Does LittleFS need the block to be erased to 0xFF for some reason (like detecting empty sectors)?

Reply:
> 

AFAIK this is not needed. It would be very bad for a sdmmc that does wear levelling because a erase would be done twice (one sector to write 0xff, another sector to write the data).

Reply:
> 

@Laczen  - I thought that erasing the sector (with 0xFF) is part of the `littlefs` internal design to clean up structures/data to increase the robustness of power failure feature.
If it is not necessary to erase the buffer, as it is used for wear leveling on raw flash, then I'm more than happy to provide a stub function for it :-).

Reply:
@Imajewski, an example of a sd/mmc configuration for littlefs can be found in https://github.com/Infineon/mtb-littlefs/blob/master/source/lfs_sd_bd.c. 
The erase just returns 0.